src/Orleans.Runtime/Catalog/ActivationData.cs

```diff
@@ -736,68 +736,64 @@ void ProcessPendingRequests()
                         }
 
                         message = _waitingRequests[i].Message;
-                        if (!MayInvokeRequest(message))
+                        try
```

Comment:
Best viewed with "Hide Whitespace" enabled, since this is just wrapping the block in a try-catch, rejecting the message which triggered the exception